base_layer/wallet_ffi/src/lib.rs

```diff
@@ -1365,6 +1422,159 @@ pub unsafe extern "C" fn completed_transaction_get_destination_public_key(
     Box::into_raw(Box::new(m))
 }
 
+/// Gets the TariExcess of a TariCompletedTransaction
+///
+/// ## Arguments
+/// `transaction` - The pointer to a TariCompletedTransaction
+/// `error_out` - Pointer to an int which will be modified to an error code should one occur, may not be null. Functions
+/// as an out parameter.
+///
+/// ## Returns
+/// `*mut TariExcess` - Returns the transaction excess, note that it will be
+/// ptr::null_mut() if transaction is null, if the transaction status is Pending, or if the number of kernels is not
+/// exactly one.
+///
+/// # Safety
+/// The ```excess_destroy``` method must be called when finished with a TariExcess to prevent a memory leak
+#[no_mangle]
+pub unsafe extern "C" fn completed_transaction_get_excess(
+    transaction: *mut TariCompletedTransaction,
+    error_out: *mut c_int,
+) -> *mut TariExcess
+{
+    let mut error = 0;
+    ptr::swap(error_out, &mut error as *mut c_int);
+    if transaction.is_null() {
+        error = LibWalletError::from(InterfaceError::NullError("transaction".to_string())).code;
+        ptr::swap(error_out, &mut error as *mut c_int);
+        return ptr::null_mut();
+    }
+
+    // check the tx is not in pending state
+    if matches!((*transaction).status, TransactionStatus::Pending) {
```

Comment:
I just realized that you must also check if it has the Imported status, that means a UTXO was imported and faux transaction was created to document the event but it wont have a formal transaction behind it.

Reply:
Thanks Phil - I'll add that `Imported` status